### ISSUE TITLE
Nextcloud app installer says pdlib is missing even though test suite runs without error

### ISSUE THREAD

**Issue by lsumner4** [matiasdelellis/facerecognition#399]
Hey, Thanks for reporting issues back to Nextcloud Face Recognition. Please, try to complete this report in detail so we can help you easier. :smile:

Make sure you read all the [documentation](https://github.com/matiasdelellis/facerecognition/wiki), and the [FAQ](https://github.com/matiasdelellis/facerecognition/wiki/FAQ), and that the issue has not been reported before. :wink:

### Expected behaviour

App installs on nextcloud after pdlib is configured.

### Actual behaviour

Application installer complains that pdlib library is missing and therefore cannot enable the app.

### Steps to reproduce
1. Using Ubuntu 20.04 install the pdlib from the repo.
2. Attempt to the install the face recognition application from nextcloud app store.
3.

### Server configuration

* **Operating system:**
Ubuntu 20.04 LTS

* **Pdlib version:**
Version provided by "easy" installer

* **How is DLib installed:** Make sure it is working correctly with this [tool](https://github.com/matiasdelellis/pdlib-min-test-suite)

* **How is PDlib installed:** Make sure it is working correctly with this [tool](https://github.com/matiasdelellis/pdlib-min-test-suite)
Via the provided "easy" installer for Focal

* **PHP version:**
7.4.3

* **Web server:**
Apache

* **Database:**
MySQL

* **Nextcloud version:**
20.03

### Client configuration

* **Browser:**

* **Operating system:**

#### Background task log with debug.
<details>
<summary>sudo -u apache php occ -vvv face:background_job</summary>

```
Insert your background log here
```
</details>

#### Web server error log
<details>
<summary>Web server error log</summary>

```
Insert your webserver log here
```
</details>

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>

```
Insert your Nextcloud log here
```
</details>

#### Browser log
<details>
<summary>Browser log</summary>

```
Insert your browser log here, this could for example include:

a) The javascript console log
b) The network log
c) ...
```
</details>

**Comment by DruculaC** [comment]
Me too.

**Comment by DruculaC** [comment]
Operating system: Ubuntu 20.04 LTS
Background task log with debug:
`In Application.php line 598:

  [Symfony\Component\Console\Exception\NamespaceNotFoundException]
  There are no commands defined in the "face" namespace.


Exception trace:
  at /var/www/nextcloud/3rdparty/symfony/console/Application.php:598
 Symfony\Component\Console\Application->findNamespace() at /var/www/nextcloud/3rdparty/symfony/console/Application.php:651
 Symfony\Component\Console\Application->find() at /var/www/nextcloud/3rdparty/symfony/console/Application.php:236
 Symfony\Component\Console\Application->doRun() at /var/www/nextcloud/3rdparty/symfony/console/Application.php:148
 Symfony\Component\Console\Application->run() at /var/www/nextcloud/lib/private/Console/Application.php:215
 OC\Console\Application->run() at /var/www/nextcloud/console.php:100
 require_once() at /var/www/nextcloud/occ:11
`

**Comment by matiasdelellis** [comment]
Hi @lsumner4 

> How is PDlib installed: Make sure it is working correctly with this tool
> Via the provided "easy" installer for Focal

The php configuration for the command line (where the test runs), is different from where the server runs (Apache, nginx, etc). Even if you use the easy version, you must make sure that the configuration is valid for your server.

So, please, check your /etc/php*** folders. If you have an 'cli', 'fpm', 'apache' folder or anything else, remember add the pdlib.ini file ti enable the extension to the server.

**Comment by DruculaC** [comment]
Yes, as you mentioned, I have copy pdlib.so to php/mods-available, then phpenmod pdlib, it turns allright now.

> 
> 
> Hi @lsumner4
> 
> > How is PDlib installed: Make sure it is working correctly with this tool
> > Via the provided "easy" installer for Focal
> 
> The php configuration for the command line (where the test runs), is different from where the server runs (Apache, nginx, etc). Even if you use the easy version, you must make sure that the configuration is valid for your server.
> 
> So, please, check your /etc/php*** folders. If you have an 'cli', 'fpm', 'apache' folder or anything else, remember add the pdlib.ini file ti enable the extension to the server.

Yes, as you mentioned, I have copy pdlib.so to php/mods-available, then phpenmod pdlib, it turns allright now.

**Comment by lsumner4** [comment]
Where is the pdlib.so file saved by default? I get what you're saying about the php version and I'm sure that's likely the case. I'll try to copy the configuration over.

**Comment by lsumner4** [comment]
So phpenmod pdlib allowed me to install the app on nextcloud but now the models are missing when I tried to run the OCC background_job command. I think the "easy" setup just doesn't have enough explanation to actually get to a functioning application. I'll admit I'm a casual tinkerer at best, but I've dealt with nextcloud a fair bit and this is probably the hardest app to install I've encountered thus far. The wiki doesn't really explain the installation steps even for the manual install in enough detail.

I know it's still in-progress, but just some feedback from a more casual user.

**Comment by matiasdelellis** [comment]
@lsumner4 
> So phpenmod pdlib allowed me to install the app on nextcloud but now the models are missing when I tried to run the OCC background_job command. I think the "easy" setup just doesn't have enough explanation to actually get to a functioning application. I'll admit I'm a casual tinkerer at best, but I've dealt with nextcloud a fair bit and this is probably the hardest app to install I've encountered thus far.

haha.. :see_no_evil: 
Unfortunately you are obviously right. :sweat_smile: 

> The wiki doesn't really explain the installation steps even for the manual install in enough detail.

https://github.com/matiasdelellis/facerecognition#configuration
https://github.com/matiasdelellis/facerecognition/wiki/Installation#install-facerecognition-app

In both places it indicates that you must install the models !.

> I know it's still in-progress, but just some feedback from a more casual user.

Ok.. Comments are always welcome, I will try to improve the wiki :wink:

**Comment by Rai2023** [comment]
After some tweaks I was able to compile the pdlib for my raspi4 (raspian bullseye). Configured every php.ini I found on it, copied the pdlib.so to place, I assume they were the right ones, ra the test suite. Result: Cli version is good. But the fpm module doesn't show up in phpinfo(). 
I assume, that it compiled for php version 7.4 and fpm uses php version 8.0, referring to phpinfo().
How to fix that?
